### PR TITLE
KAFKA-15364: Replay BrokerRegistrationChangeRecord.logDirs

### PR DESCRIPTION
**Pull request by soarez**
Any directory changes must be considered when replaying BrokerRegistrationChangeRecord. This is necessary
to persist directory failures in the cluster metadata, which #14902 missed.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by soarez** [comment]
> 36 tests have failed There are 0 new tests failing, 36 existing failing and 271 skipped.

@rondagostino PTAL

**Review by OmniaGM — APPROVED** [review]
Seems like a straightforward fix. LGTM.

**Review by viktorsomogyi — APPROVED** [review]
LGTM, it fixes the failures in #14977 but I'll leave this to @rondagostino to merge this if he doesn't have any more comments.

**Comment by viktorsomogyi** [comment]
@stanislavkozlovski I think this is important for 3.7. I now restarted the unit tests as previously seems like Jenkins terminated. I'll make sure to cherry-pick this to the 3.7 branch too and the tests there as well.

**Comment by viktorsomogyi** [comment]
Raised the cherr-pick in #15042

**Review by rondagostino — APPROVED** [review]
LGMT assuming test pass -- thanks!

**Comment by rondagostino** [comment]
`34 tests have failed.  There are 26 new tests failing, 8 existing failing and 288 skipped.`

`shouldHaveSamePositionBoundActiveAndStandBy –  org.apache.kafka.streams.integration.ConsistencyVectorIntegrationTest` failed in 3 of the 4 JDK versions (8, 11, and 17), but that seems unrelated.  `testTaskRequestWithOldStartMsGetsUpdated() – org.apache.kafka.trogdor.coordinator.CoordinatorTest` failed in JDK 21 and 11 and also seems unrelated.

@viktorsomogyi WDYT?

**Comment by rondagostino** [comment]
`shouldHaveSamePositionBoundActiveAndStandBy` only failed on JDKs 8 and 21 on the 3.7 PR (https://github.com/apache/kafka/pull/15042), which overall had `37 tests have failed.  There are 0 new tests failing, 37 existing failing.`

`testTaskRequestWithOldStartMsGetsUpdated` only failed on JDK 11 on the 3.7 PR.

I'm inclined to say these are flaky tests.  I'mm ognna merge this and the other one.

**Pull request merged**
